File: Weapon/ColdWeapon/ColdWeapon.php
<?php

namespace Weapon;


abstract class ColdWeapon extends Weapon {
    protected $bladeLength;
    protected $material;

    final public function getMaterial(){
        return $this->material;
    }

    final public function getBladeLength(){
        return $this->bladeLength;
    }
    final public function setBladeLength(float $bladeLength){
        $this->bladeLength = $bladeLength;
    }
    final public function setMaterial(string $material){
        $this->material = $material;
    }

}

File: Weapon/ColdWeapon/ColdWeapon.py
import sys
sys.path.append('../')
from Weapon import *


class ColdWeapon(Weapon):
    __bladeLength = None
    __material = None

    @property
    def blade_length(self):
        return self.__bladeLength

    @blade_length.setter
    def blade_length(self, value: float):
        self.__bladeLength = value

    @property
    def material(self):
        return self.__material

    @material.setter
    def material(self, value: str):
        self.__material = value
